turbinia/workers/analysis/linux_acct.py

```diff
@@ -0,0 +1,126 @@
+
+
+class LinuxAccountAnalysisTask(TurbiniaTask):
+  """Task to analyze a Linux password file."""
+
+  REQUIRED_STATES = [state.ATTACHED, state.MOUNTED]
+
+  def run(self, evidence, result):
+    """Run the Linux Account worker.
+    Args:
+        evidence (Evidence object):  The evidence to process
+        result (TurbiniaTaskResult): The object to place task results into.
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    # Where to store the resulting output file.
+    output_file_name = 'linux_account_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # What type of evidence we should output.
+    output_evidence = ReportText(source_path=output_file_path)
+
+    try:
+      collected_artifacts = extract_artifacts(
+          artifact_names=['LoginPolicyConfiguration'],
+          disk_path=evidence.local_path, output_dir=self.output_dir)
+    except TurbiniaException as e:
+      result.close(self, success=False, status=str(e))
+      return result
+
+    for filepath in collected_artifacts:
+      if not filepath.endswith('shadow'):
+        continue
+
+      shadow_file = []
+      # Read the input file
+      with open(filepath, 'r') as input_file:
+        shadow_file = input_file.readlines()
+
+      hashnames = self._extract_linux_credentials(shadow_file)
+      (report, priority, summary) = self.analyse_shadow_file(
+          shadow_file, hashnames)
+      output_evidence.text_data = report
+      result.report_priority = priority
+      result.report_data = report
+
+      # Write the report to the output file.
+      with open(output_file_path, 'wb') as fh:
+        fh.write(output_evidence.text_data.encode('utf-8'))
+
+      # Add the resulting evidence to the result object.
+      result.add_evidence(output_evidence, evidence.config)
+      result.close(self, success=True, status=summary)
+      return result
+    result.close(self, success=True, status='No shadow files found')
+    return result
+
+  @staticmethod
+  def _extract_linux_credentials(shadow):
+    """Extract credentials from a Linux shadow files.
+    Args:
+      shadow (list): shadow file contents (list of str).
+    Returns:
+      dict: of hash against username.
+    """
+    hashnames = {}
+    for line in shadow:
+      (username, passwdhash, _) = line.split(':', maxsplit=2)
+      hashnames[passwdhash] = username
+    return hashnames
+
+  def analyse_shadow_file(self, shadow, hashes):
+    """Analyses a Linux shadow file.
+    Args:
+      shadow (list): shadow file content (list of str).
+      hashes (dict): dict of hashes to usernames
+    Returns:
+      Tuple(
+        report_text(str): The report data
+        report_priority(int): The priority of the report (0 - 100)
+        summary(str): A summary of the report (used for task status)
+      )
+    """
+    report = []
+    summary = 'No weak passwords found'
+    priority = Priority.LOW
+    weak_passwords = bruteforce_password_hashes(shadow)
```

Comment:
One thing that would be cool to do with this data is save the weak passwords to a separate file, and then later it would be easy to take a one-liner script to gather all of those files from the saved task output.  Then we could re-use that data either in Turbinia or in our other tools.

Probably the easiest way to do that is to create a new Evidence object type (I think it could just extend `TextFile` with a `pass` similar to the `FilteredTextFile`).    You could always do this in a later PR  too (or just defer and file an issue).

Reply:
Roger, will file an issue